### ISSUE TITLE
SSL certification error

### ISSUE THREAD

**Issue by asnagni** [D7EAD/liboai#51]
### Describe the bug

Hello guys,
	First of all I would like to say thank you for this work and providing to the community this c++ Library 👍.
	I was on OpenAI API website looking for a C++ option and I came across your library. I was able to easily build it and integrate it into our project without any issue (Android & iOS). Our code is C++ based - obviously 😊.
	To make the learning curve simple, I just took your “conversasion” example and used it without any modification. Just to make sure that everything is working fine. Unfortunatunatly the “ChatCompletion” call is failling. This is the error/exception message that I have:
	liboai::netimpl::Session::Perform(): Peer certificate cannot be authenticated with given CA certificates (E_CURLERROR:0x06)

	
	Based on your examples/tutorials I’m not seeing anywhere where you need to:
	1) provide a certificate file. If this is needed and it omitted in the example could please tell which Api call to use. 

	2) or the need to turn of the SSL verification - to do something like this:
		curl_ = curl_easy_init();
		curl_easy_setopt(curl_, CURLOPT_SSL_VERIFYPEER, 0L);

	Would you know how to fix this issue?

Thank you,
Stay safe

### To Reproduce

1. Build current version for your iOS or Android project. There is nothing special if your have Curl & Json 
2. Use the conversation code/example to test
 

### Code snippets

```C++
liboai::OpenAI oai;
        
        // create a conversation
        liboai::Conversation convo;

        // add a message to the conversation
        convo.AddUserData("Hello, how are you?");

        if (oai.auth.SetKey(m_openaiApiKey))
        {
            try
            {
                liboai::Response response = oai.ChatCompletion->create( "gpt-3.5-turbo", convo );

                // update our conversation with the response
                convo.Update(response);
            
                // print the response
                std::cout << convo.GetLastResponse() << std::endl;
            }
            catch (std::exception& e)
            {
                std::cout << e.what() << std::endl;
            }
        }
```

### OS

iOS

### Library version

liboai v3.2.1

**Comment by asnagni** [comment]
Hi guys,
   I have an additional information. I downloaded the single header file library "openai-cpp" and it is working. I don't have the ssl certificate issue. Which may indicate that the issue is with liboai.
   I have and additional information that may help find the issue, I'm using curl-8.1.2
   
Thank you,
Stay safe

**Comment by D7EAD** [comment]
Hi,

Sorry for the late response.

In <code>liboai</code>'s core network back-end source code, I do not force the library to verify certificate of peers. The default for this setting, according to cURL, is enabled - so peer certificates will be verified (see [here](https://curl.se/libcurl/c/CURLOPT_SSL_VERIFYPEER.html)).

I can easily add an option to enable or disable this, and this will fix your issue. I can easily do this in due time, and once done, I will update here.

Best.

**Comment by asnagni** [comment]
Hi there,
	No worries for the delay. Thank you very much for your reply, I do appreciate. Yes I think it would be a useful addition to the liboai library. Please let me know when you have it in so I can remove the patch that I have :-). With the patch that I have implemented it is working now.

Thank you,
Stay safe

**Comment by D7EAD** [comment]
Hi,

I have released a quick-fix PR (#52) for this issue. Please use the new pre-processor definition described in the PR, as well as the debug pre-processor definition `LIBOAI_DEBUG `to ensure the peer verification is being disabled on your end (see line 2 in the terminal output below).

![image](https://github.com/D7EAD/liboai/assets/42398281/6f5549f0-6d79-4707-b67a-b911aba73a32)

Let me know the results.

Best.

**Comment by asnagni** [comment]
Hi,
	Sorry for the late response but I had so many things to deal with that this went at a lower priority. Thank you for the info and I will look into today.


Thank you again,
Stay safe
